//
// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package router;

import "engine/datasource/datasource.proto";
import "api/status.proto";
import "api/common.proto";

option cc_generic_services = true;

service RouterService {
  rpc Route(RouteRequest) returns (RouteResponse);
}

message RouteRequest {
  message Table {
    string db = 1;
    string table = 2;
  }
  scql.pb.RequestHeader header = 1;
  repeated Table tables = 2;
}

message RouteResponse {
  scql.pb.Status status = 1;
  repeated string datasource_ids = 2;
  // datasource_ids --> DataSouorce map
  map<string, scql.engine.DataSource> datasources = 3;
}